extensions/migadu/src/aliases.tsx

```diff
@@ -24,7 +24,7 @@ export default function Aliases() {
 
 function AliasesIndex({ domain }: { domain: string }) {
   const { push } = useNavigation();
-  const [aliases, setAliases] = useState<Alias[]>([]);
```

Comment:
Suggestion (non-blocking): In an effort of pure refactoring, we can use `useCachedPromise` hook (instead of `useCachedState`) to replace the whole `getDomainAliasesFromApi()`, `isLoading`, `error` etc. and we won't need `useCachedState`. So something like:

```typescript
const {isLoading, error, mutate, data: aliases} = useCachedPromise(getDomainAliases, [domain], {failureToastOptions: {title: "Error while getting domain aliases"}});
```

This will require some refactoring in callApi: removing the failure toast there and bubbling up the error msg as part of `Error` instead of `ErrorResponse`.

This way we can also optimistically mutate based on deletion etc.

This refactoring can be applied to most commands where we are using `useCachedState` in this PR. If this is intrusive for this PR, it can be handled as part of a separate PR too. What do you think?